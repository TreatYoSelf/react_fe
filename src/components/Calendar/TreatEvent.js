import React from 'react';
import { StyleSheet, Text, View } from 'react-native';

export default function TreatEvent({ title, duration, time }) {
    return (
        <View>
            <Text style={styles.header}>{parseInt(time) < 12 ? `${time} AM` : `${time % 12} PM`}</Text>
            <View style={styles.container}>
                <Text style={styles.header}>{title}</Text>
                <Text style={styles.header}>{`${duration} Mins`}</Text>
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        alignItems: 'center',
        justifyContent: 'space-around',
        flexDirection: 'row',
        width: 300,
        padding: 10,
        backgroundColor: "#e6f7ff",
        borderWidth: 5,
        borderColor: '#003045',
        margin: 10
    },
    header: {
        fontWeight: 'bold',
        color: '#003045'
    },
    selected: {
        backgroundColor: "blue",
    }
});