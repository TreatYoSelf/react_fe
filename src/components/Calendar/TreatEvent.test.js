import React from 'react';
import { act, create } from 'react-test-renderer';

import TreatEvent from './TreatEvent';

describe('<TreatEvent />', () => {
    it('renders correctly', async () => {
        let tree;
        const amProps = {
            time: 10,
            title: 'testEvent',
            duration: 75,
        }

        const pmProps = {
            time: 16,
            title: 'testEvent',
            duration: 75,
        }

        act(() => {
            tree = create(<TreatEvent {...amProps} />);
        })
        expect(tree.toJSON()).toMatchSnapshot();

        act(() => {
            tree.update(<TreatEvent {...pmProps} />)
        })
        expect(tree.toJSON()).toMatchSnapshot();
    });
});