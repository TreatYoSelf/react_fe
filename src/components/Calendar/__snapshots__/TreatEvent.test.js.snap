// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreatEvent /> renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#003045",
        "fontWeight": "bold",
      }
    }
  >
    10 AM
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e6f7ff",
        "borderColor": "#003045",
        "borderWidth": 5,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "margin": 10,
        "padding": 10,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#003045",
          "fontWeight": "bold",
        }
      }
    >
      testEvent
    </Text>
    <Text
      style={
        Object {
          "color": "#003045",
          "fontWeight": "bold",
        }
      }
    >
      75 Mins
    </Text>
  </View>
</View>
`;

exports[`<TreatEvent /> renders correctly 2`] = `
<View>
  <Text
    style={
      Object {
        "color": "#003045",
        "fontWeight": "bold",
      }
    }
  >
    4 PM
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#e6f7ff",
        "borderColor": "#003045",
        "borderWidth": 5,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "margin": 10,
        "padding": 10,
        "width": 300,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#003045",
          "fontWeight": "bold",
        }
      }
    >
      testEvent
    </Text>
    <Text
      style={
        Object {
          "color": "#003045",
          "fontWeight": "bold",
        }
      }
    >
      75 Mins
    </Text>
  </View>
</View>
`;
